client/components/SendZoopDialog.tsx

```diff
@@ -0,0 +1,155 @@
+import React, {
+    useState,
+    useEffect 
+} from "react";
+import {useNavigate} from "react-router-dom";
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText"
+import DialogActions from "@mui/material/DialogActions";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Link from "@mui/material/Link";
+import Tooltip from "@mui/material/Tooltip";
+import Fab from "@mui/material/Fab";
+import SendIcon from "@mui/icons-material/Send";
+import { 
+    usePostZoopMutation, 
+    useGetAllUsersQuery,
+    useGetMeQuery
+}  from "../features/api";
+import { useSelector } from "react-redux";
+import { RootState} from "../app/store";
+import type { User } from '../../src/types/custom'
+
+const SendZoopDialog = () => {
+    const [open, setOpen] = useState(false);
+    const [username, setUsername] = useState("");
```

Comment:
```suggestion
    const [selectedUser, setSelectedUser] = useState<User | null>(null);
```

Reply:
Is meant to be the user who is receiving the Zoop?

Reply:
Probably a stupid question, but why does this need to be stored in state?

Reply:
Because it is a piece of data inside your form. You want to allow the person to select a user in the form and then submit a zoop to them, so you need to track which user they selected so you know which user ID to send in the sendZoop mutation

Reply:
I was using the username string from the Dialog textfield to compare to the `User` objects and find the right user ID to use in the Zoop for the `receiverId`. Wouldn't storing the selected user in state be an unnecessary additional step?

Reply:
But it won't be a typed `User` object, will it? The data I am getting from the Autocomplete field of the dialog is one of the objects in the array of options (each option is a type `{label: string, id: number}`). I think the `id` property on those options should match the IDs for the same users in the DB. So `selectedUser` would just be one of those option objects, right? And I could just use `selectedUser.id` as the `receiverId` on the Zoop, right? Or do you want me to find the `User` object in `usersData` and store it in the component's state?

Reply:
Yes, the ID in Autocomplete should match the ID of the user in the DB. So selectedUser should be whatever Autocomplete expects (can't remember if the selected value is the `{label: String, id: ID}` object or if it's just the ID). And then you will have the ID saved in your state so you can reference it to make the mutation.

The docs for Autocomplete should have some guidance for how to store the state correctly, from the examples it looks like it's probably a new object that has label & id - but either way, that means you can extract the ID from state: https://mui.com/material-ui/react-autocomplete

Reply:
Ahhhhh, I didn't realize you wanted me to make a new mutation. Can do. Makes sense.

Reply:
There shouldn't need to be a new mutation other than `sendZoop`? You should be able to call `sendZoop` with the selected user ID